wgengine/tstun/tun.go

```diff
@@ -204,6 +205,7 @@ func (t *TUN) poll() {
 
 func (t *TUN) filterOut(buf []byte) filter.Response {
 	p := t.parsedPacketPool.Get().(*packet.ParsedPacket)
+	defer t.parsedPacketPool.Put(p)
 	p.Decode(buf)
```

Comment:
There are a bunch of fields in ParsedPacket that Decode doesn't necessarily clear in all paths. Should you do a `*p = packet.ParsedPacket{}` first to zero it out?

Reply:
I believe (though I may have missed something :/) that all the fields that are relevant for a given `IPProto` are always cleared. So e.g. the ports may contain garbage for a `Unknown` packet, but no `Unknown` packet should ever have its ports inspected anyway. I guess I should document this.

Reply:
I'm not sure if clearing those fields where they don't make sense is actually desirable, since that code is inherently relying on meaningless data.